src/V1/FabricationFile.php

```diff
@@ -155,4 +156,29 @@ public function setDirectoryPath(string $DirectoryPath): FabricationFileInterfac
 
         return $this;
     }
+
+    public function getPreferredTemplateTrees(): array
+    {
+        if ($this->PreferredTemplateTrees === null) {
+            throw new \LogicException('FabricationFile PreferredTemplateTrees has not been set.');
+        }
+
+        return $this->PreferredTemplateTrees;
+    }
+
+    public function hasPreferredTemplateTrees(): bool
+    {
+        return $this->PreferredTemplateTrees !== null;
+    }
+
+    public function setPreferredTemplateTrees(string ...$PreferredTemplateTrees): FabricationFileInterface
```

Comment:
consider modeling Preferred Template Trees as a first class DAO

Reply:
Hrmm... I think I have some thoughts on how to do that, but I might tackle that as as a follow-up refactor to the full thing if you're cool with that.

Reply:
Totally cool with that

Reply:
Great. Created [BUPH-113](https://55places.atlassian.net/browse/BUPH-113)